### PR TITLE
Gen 1 iOS polling fix

### PR DESCRIPTION
**Pull request by jaredperreault-okta**
## PR Checklist

- [ ] Have you verified the basic functionality for this change?
- [ ] Did you add tests, as appropriate, following our [Automated Test guidelines](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2676497890/Automated+Testing+in+the+Signin+Widget)?
- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?
- [ ] Did you verify the change by running [downstream monolith artifact](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/102897979/Sign-in+Widget+Development#Sign-inWidgetDevelopment-Instructionstocreateandrunthedownstreamartifact(d16t))? (Provide link to build below)
- [ ] Does this PR include noticeable changes to the UI? (If yes, attach screenshots/video below)

### Issue:

- OKTA-823470

### Screenshot/Video:

https://github.com/user-attachments/assets/6732a6d4-b25e-43a7-8191-d40671b19455

https://github.com/user-attachments/assets/7b15cb03-c2ed-4c02-bda6-0d047cf4468c

### CODE REVIEW

**Review by lesterchoi-okta — COMMENTED** [review]
Re-target to `7.26` branch

**Review by stevelind-okta — COMMENTED** [review]
Hi, not sure if this was ready for my review but joining in at @lesterchoi-okta 's request. 

Let's make sure to document the risk assessment (high) and document the testing done, which I believe was extensive. It would be great to see a video of the fix working on some of the devices which were tested.

**Review by lesterchoi-okta — APPROVED** [review]

File: test/unit/jest/jest-setup.js [hunk 1]
```diff
@@ -15,6 +15,14 @@ global.getJasmineRequireObj = function jestSetupGlobalJasmine() {
...
 };
 global.jasmine = jasmine;
+if (global.document) {
+  let docHidden = false;
+  Object.defineProperty(global.document, 'hidden', {
+    configurable: true,
+    get() { return docHidden; },
+    set(bool) { docHidden = Boolean(bool); }
+  });
+}
```
[the remark this answers is not in the document]
Comment (jaredperreault-okta) [reply test/unit/jest/jest-setup.js:25]: Yes. Otherwise I cannot manually toggle the value during tests

**Comment by jaredperreault-okta** [comment]
@stevelind-okta I added some videos and updated the ticket with the Risk Level and testing performed

**Pull request merged**